test/e2e/onCLS-test.js

```diff
@@ -832,6 +832,28 @@ describe('onCLS()', async function () {
     assert.strictEqual(cls1_1.navigationType, cls2_3.navigationType);
   });
 
+  it('reports on document visibility change handling', async function () {
```

Comment:
I think the description in this test is misleading. The library always reported on `visibilitychange`, and there are already numerous tests for that behavior.

What's different here is your testing to ensure that if a developer chooses to implement their own batching (following the example in the README), that all of the relevant metric callbacks have run before their code does. So I'd rename this to clarify that.

Maybe something like: "it reports before developer-added visibilitychange listeners run"

Reply:
I changed it to "reports on batch reporting using document.visibilitychange" cause the key thing was the `document` bit.